normalizer_test.go

```diff
@@ -0,0 +1,500 @@
+			},
+		},
+		{
+			input: `-- Single line comment
+			-- Another single line comment
+			-- Another another single line comment
+			GRANT USAGE, DELETE ON SCHEMA datadog TO datadog`,
+			want: "GRANT USAGE, DELETE ON SCHEMA datadog TO datadog",
+			normalizedInfo: NormalizedInfo{
+				Tables:   []string{},
+				Comments: []string{"-- Single line comment", "-- Another single line comment", "-- Another another single line comment"},
+				Commands: []string{"GRANT", "DELETE"},
+			},
+		},
+		{
+			input: `-- Testing table value constructor SQL expression
+			SELECT * FROM (VALUES (?, ?)) AS d (id, animal)`,
+			want: "SELECT * FROM ( VALUES ( ? ) ) ( id, animal )",
+			normalizedInfo: NormalizedInfo{
+				Tables:   []string{},
+				Comments: []string{"-- Testing table value constructor SQL expression"},
+				Commands: []string{"SELECT"},
+			},
+		},
+		{
+			input: `ALTER TABLE tabletest DROP COLUMN columna`,
+			want:  "ALTER TABLE tabletest DROP COLUMN columna",
+			normalizedInfo: NormalizedInfo{
+				Tables:   []string{"tabletest"},
+				Comments: []string{},
+				Commands: []string{"ALTER", "DROP"},
+			},
+		},
+		{
+			input: `REVOKE ALL ON SCHEMA datadog FROM datadog`,
+			want:  "REVOKE ALL ON SCHEMA datadog FROM datadog",
+			normalizedInfo: NormalizedInfo{
+				Tables:   []string{"datadog"},
+				Comments: []string{},
+				Commands: []string{"REVOKE"},
+			},
+		},
+		{
+			input: "/* Testing explicit table SQL expression */ WITH T1 AS (SELECT PNO , PNAME , COLOR , WEIGHT , CITY FROM P WHERE CITY = ?), T2 AS (SELECT PNO, PNAME, COLOR, WEIGHT, CITY, ? * WEIGHT AS NEW_WEIGHT, ? AS NEW_CITY FROM T1), T3 AS ( SELECT PNO , PNAME, COLOR, NEW_WEIGHT AS WEIGHT, NEW_CITY AS CITY FROM T2), T4 AS ( TABLE P EXCEPT CORRESPONDING TABLE T1) TABLE T4 UNION CORRESPONDING TABLE T3",
+			want:  "WITH T1 AS ( SELECT PNO, PNAME, COLOR, WEIGHT, CITY FROM P WHERE CITY = ? ), T2 AS ( SELECT PNO, PNAME, COLOR, WEIGHT, CITY, ? * WEIGHT, ? FROM T1 ), T3 AS ( SELECT PNO, PNAME, COLOR, NEW_WEIGHT, NEW_CITY FROM T2 ), T4 AS ( TABLE P EXCEPT CORRESPONDING TABLE T1 ) TABLE T4 UNION CORRESPONDING TABLE T3",
+			normalizedInfo: NormalizedInfo{
+				Tables:   []string{"P", "T1", "T2", "T4", "T3"},
+				Comments: []string{"/* Testing explicit table SQL expression */"},
+				Commands: []string{"SELECT"},
+			},
+		},
+		{
+			// truncated
+			input: "SELECT * FROM users WHERE id =",
+			want:  "SELECT * FROM users WHERE id =",
+			normalizedInfo: NormalizedInfo{
+				Tables:   []string{"users"},
+				Comments: []string{},
+				Commands: []string{"SELECT"},
+			},
+		},
+	}
+
+	normalizer := NewSQLNormalizer(&SQLNormalizerConfig{
+		CollectComments: true,
+		CollectCommands: true,
+		CollectTables:   true,
+		KeepSQLAlias:    false,
+	})
+
+	for _, test := range tests {
+		t.Run("", func(t *testing.T) {
+			got, normalizedInfo, err := normalizer.Normalize(test.input)
+			if err != nil {
+				t.Errorf("error during normalization: %v", err)
+			}
```

Comment:
Not that I want to suggest bloating dependencies but given how commonplace [testify](https://pkg.go.dev/github.com/stretchr/testify() is, I'd say it would be a reasonable dependency to use since it would make this way more compact and readable (`assert.Equal(t, test.expected, actual)`).

And also, a test dependency doesn't get added to the user's dependencies so that's pretty harmless.

Reply:
I thought about adding `testify` but want to keep my `zero deps` promise. But I'm easily convinced 😄 .

Reply:
Well, we can confirm but the test dependencies shouldn't be visible to users of the package so that might still meet the requirement if we consider the dependencies to be what users see?

Reply:
I have confirmed `testify` does not get pulled in if I just install the package
```
 ~/go/src/github.com/DataDog/playground-go/ go list -json
{
        "Dir": "/Users/zhengda.lu/go/src/github.com/DataDog/playground-go",
        "ImportPath": "main",
        "Name": "main",
        "Doc": "nolint: govet",
        "Target": "/Users/zhengda.lu/go/bin/main",
        "Root": "/Users/zhengda.lu/go/src/github.com/DataDog/playground-go",
        "Module": {
                "Path": "main",
                "Main": true,
                "Dir": "/Users/zhengda.lu/go/src/github.com/DataDog/playground-go",
                "GoMod": "/Users/zhengda.lu/go/src/github.com/DataDog/playground-go/go.mod",
                "GoVersion": "1.20"
        },
        "Match": [
                "."
        ],
        "DefaultGODEBUG": "panicnil=1",
        "Stale": true,
        "StaleReason": "build ID mismatch",
        "GoFiles": [
                "main.go"
        ],
        "Imports": [
                "fmt",
                "github.com/DataDog/go-sqllexer"
        ],
        "Deps": [
                "bytes",
                "errors",
                "fmt",
                "github.com/DataDog/go-sqllexer",
                "internal/abi",
                "internal/bytealg",
                "internal/coverage/rtcov",
                "internal/cpu",
                "internal/fmtsort",
                "internal/goarch",
                "internal/godebugs",
                "internal/goexperiment",
                "internal/goos",
                "internal/itoa",
                "internal/oserror",
                "internal/poll",
                "internal/race",
                "internal/reflectlite",
                "internal/safefilepath",
                "internal/syscall/execenv",
                "internal/syscall/unix",
                "internal/testlog",
                "internal/unsafeheader",
                "io",
                "io/fs",
                "math",
                "math/bits",
                "os",
                "path",
                "reflect",
                "regexp",
                "regexp/syntax",
                "runtime",
                "runtime/internal/atomic",
                "runtime/internal/math",
                "runtime/internal/sys",
                "sort",
                "strconv",
                "strings",
                "sync",
                "sync/atomic",
                "syscall",
                "time",
                "unicode",
                "unicode/utf8",
                "unsafe"
        ]
}
```

Reply:
Nice. Thanks for testing this.